Lib/subprocess.py

```diff
@@ -606,6 +606,55 @@ def getoutput(cmd):
     return getstatusoutput(cmd)[1]
 
 
+def _use_posix_spawn():
+    """Check is posix_spawn() can be used for subprocess.
+
+    subprocess requires a posix_spawn() implementation that reports properly
+    errors to the parent process, set errno on the following failures:
+
+    * process attribute actions failed
+    * file actions failed
+    * exec() failed
+    """
+    if _mswindows or not hasattr(os, 'posix_spawn'):
+        # os.posix_spawn() is not available
+        return False
+
+    if sys.platform == 'darwin':
+        # posix_spawn() is a syscall on macOS and properly reports errors
+        return True
+
+    # Check libc name and runtime libc version
+    try:
+        ver = os.confstr('CS_GNU_LIBC_VERSION')
+        # parse 'glibc 2.28' as ('glibc', (2, 28))
+        parts = ver.split(maxsplit=1)
+        if len(parts) != 2:
+            # reject unknown format
+            raise ValueError
+        libc = parts[0]
+        version = tuple(map(int, parts[1].split('.')))
+
+        if libc == 'glibc' and version >= (2, 26):
+            # glibc 2.26 added a pipe to the POSIX implementation
+            # of posix_spawn() to properly report errors to the parent process.
+            return True
```

Comment:
Note that the [patch](https://sourceware.org/git/?p=glibc.git;a=commit;h=ccfb2964726512f6669fea99a43afa714e2e6a80) that added a pipe to the POSIX implementation also switched to unconditional use of `fork`. Since the main motivation for using `posix_spawn` appears to be performance benefits of `vfork`, it seems that glibc's POSIX implementation shouldn't be used at all. I suggest to remove this branch.

Otherwise, LGTM.

Reply:
Ah. I didn't require to use vfork. But ok, I modified my PR to prefer posix_spawn() implementations which can use vfork in some cases for best performances. Let's start with a minimum platform support, and extend it later.